orderbook/src/orderbook.rs

```diff
@@ -0,0 +1,89 @@
+use model::{Order, UserOrder};
+use primitive_types::H160;
+use tokio::sync::RwLock;
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum AddOrderError {
+    AlreadyExists,
+    InvalidSignature,
+    #[allow(dead_code)]
+    PastNonce,
+    #[allow(dead_code)]
+    PastValidTo,
+}
+
+#[derive(Debug)]
+pub enum RemoveOrderError {
+    DoesNotExist,
+}
+
+#[derive(Debug, Default)]
+pub struct OrderBook {
+    // TODO: Store more efficiently (for example HashMap) depending on functionality we need.
+    pub orders: RwLock<Vec<Order>>,
```

Comment:
nit: Even for the current functionality (duplicate check, remove) a HashSet might be more appropriate